src/librustdoc/html/highlight.rs

```diff
@@ -957,47 +972,47 @@ impl<'src> Classifier<'src> {
     }
 
     /// Concatenate colons and idents as one when possible.
-    fn get_full_ident_path(&mut self) -> Vec<(TokenKind, usize, usize)> {
-        let start = self.byte_pos as usize;
-        let mut pos = start;
+    fn get_full_ident_path(&mut self) -> Option<usize> {
         let mut has_ident = false;
+        let mut nb_items = 0;
 
-        loop {
+        let ret = loop {
```

Comment:
We don't return in this `loop` because we need to call `stop_peeking` before exiting this function. So instead, we `break` the value to return, call `stop_peeking` and then actually return. Not super graceful but it works.